libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1117,7 +1140,7 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
     protected checkEmulatingOAuthCards(context: TurnContext): void {
         if (!this.isEmulatingOAuthCards &&
             context.activity.channelId === 'emulator' &&
-            (!this.credentials.appId || !this.credentials.appPassword)) {
```

Comment:
We don't want any more checks that depend on the existence of a password, since not all oauth client credential flows have passwords. The only common ground is the client id + some sort of credential which wildly varies in shape